### ISSUE TITLE
Returned PFUser is actually a BFTask

### ISSUE THREAD

**Issue by sherwinzadeh** [ParsePlatform/ParseTwitterUtils-iOS#42]
I'm having a weird problem that after I enter Twitter credentials, instead of getting a PFUser, I get a BFTask object. I'm coding in Swift 2.3 and while the debugger says it's a PFUser, it crashes on setValue:forKey: calls and inspecting with the debugger (po command) reveals that it's actually a BFTask. I stepped through all of the calls and I see that in the code PFTwitterPrivateUtilities.m:

`- (instancetype)pftw_continueWithMainThreadUserBlock:(PFUserResultBlock)block {

    return [self pftw_continueWithMainThreadBlock:^id(BFTask *task) {

        if (block) {

            block(task.result, task.error);

        }
        return nil;
    }];
}`

The task.result that's passed into the block callback is in fact another BFTask and not a PFUser as expected.

Any suggestions? I'm stuck on this for a long time.

One thing to note: I imported ParseTwitterUtils-iOS into my own project and I'm not using CocoaPods as I prefer Carthage instead.

Please help!!!

**Comment by sebasdeweert** [comment]
Same problem here. Did you fix this?

**Comment by sherwinzadeh** [comment]
Yes -- if I remember correctly the issue was that it was linking with an out of date version of the Bolts framework. This was a while ago so things may have changed and Carthage wasn't supported that well. There was a fork that had fixed the issue. Here's what I put in my Cartfile to fix it:

`github "BoltsFramework/Bolts-ObjC" ~> 1.6`
`github "facebook/facebook-ios-sdk"`
`github "andrey-krukovskiy/Parse-SDK-iOS-OSX" "master"`
`github "andrey-krukovskiy/ParseTwitterUtils-iOS" "master"`
`github "antigp/ParseFacebookUtils-iOS" "nlutsenko.carthage"`

Honestly, though, I would switch to CocoaPods as much as I prefer Carthage. Too many problems like this with Facebook's frameworks.
